docs/public/installation/zonemaster-backend.md

```diff
@@ -711,7 +718,69 @@ Now go back to "[Database configuration](#53-database-configuration-freebsd)"
 to create the database tables and then continue with the steps after that.
 
 
-## 9. Cleaning up the database
+## 9. Installation with Clickhouse (experimental)
+
+First follow the installation instructions for the OS in question, and then go
+to this section to install Clickhouse.
+
+
+### 9.1. Clickhouse (Rocky Linux)
+
+There is no specific installation instructions for Rocky Linux yet. In most
+parts the instructions for Debian/Ubuntu can be followed.
+
+
+### 9.2. Clickhouse (Debian/Ubuntu)
+
+> This is an experimental feature.
+
+To install Clickhouse, follow the instructions from the [official
+documentation](https://clickhouse.com/docs/en/install#available-installation-options).
+
+Configure Zonemaster::Backend to use the correct database engine:
+
+```sh
+sudo sed -i '/\bengine\b/ s/=.*/= Clickhouse/' /etc/zonemaster/backend_config.ini
+```
+
+> **Note:** See the [backend configuration] documentation for details.
+
+Install the Perl bindings (Clickhouse relies on the MySQL DBI).
+
+```sh
+sudo apt install libdbd-mysql-perl
+```
+
+To create the database and the database user (unless you keep an old database).
+Edit the command first if you want a non-default database name, user name or
+password.
+
+> **Note**: The Clickhouse MySQL interface requires a [double SHA1
+> password](https://clickhouse.com/docs/en/operations/settings/settings-users#password_double_sha1_hex)
+> ```
+> PASSWORD=$(base64 < /dev/urandom | head -c8); echo "$PASSWORD"; echo -n "$PASSWORD" | sha1sum | tr -d '-' | xxd -r -p | sha1sum | tr -d '-'
+> ```
+
+```sh
+clickhouse-client --ask-password -q "CREATE USER zonemaster IDENTIFIED WITH DOUBLE_SHA1_HASH BY '<DOUBLE_SHA1_PASSWORD>';"
+clickhouse-client --ask-password -q "CREATE DATABASE zonemaster;"
+clickhouse-client --ask-password -q "GRANT CREATE TABLE, DROP TABLE, SELECT, INSERT, ALTER UPDATE ON zonemaster.* TO zonemaster;"
+```
+
+Update the `/etc/zonemaster/backend_config.ini` file with database name, username
+and password if non-default values are used.
+
+Now go back to "[Database configuration](#43-database-configuration-debianubuntu)"
+to create the database tables and then continue with the steps after that.
+
+
+### 9.3. Clickhouse (FreeBSD)
+
+There is no specific installation instructions for FreeBSD yet. In most parts
+the instructions for Debian/Ubuntu can be followed.

```

Comment:
There is a [FreeBSD port](https://www.freshports.org/databases/clickhouse/) for Clickhouse, so installing it is a `pkg install -y clickhouse` away.

You will also need to do `pkg install -y p5-DBD-MySQL` and adjust the `sed` command so that it modifies `/usr/local/etc/zonemaster/backend_config.ini` instead of `/etc/zonemaster/backend_config.ini`.